File: Stacks_Queues/simplify_path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stack, path = [], path.split("/")
        
        for current in path:
            if stack and current == "..":
                stack.pop()
            elif current not in (".", "", ".."):
                stack.append(current)
                
        return "/" + "/".join(stack)

File: Stacks_Queues/simplify_path.rb
#https://leetcode.com/problems/simplify-path/

def simplify_path(path)
    
    canonical_path = []
    directories = path.split('/')

    for dir in directories
        if dir == ".."
            canonical_path.pop
        elsif dir == "."
            next
        else
            canonical_path << dir if !dir.empty?
        end
    end
    "/#{canonical_path.join('/')}" 
end
